### PR TITLE
Fix text rendering by correctly calling `onPreDraw` before onDraw

### PR DESCRIPTION
**Pull request by geoff-powell**
[This PR](https://github.com/cashapp/paparazzi/pull/170/files) added the reflection call to force `mFirstFrameExecuted` to be true due to some animation issues. This seems to have caused issues with the changes made in Iguana around rendering which calls `onPreDraw` after render. 

A[ similar PR](https://github.com/cashapp/paparazzi/pull/1380) fixed an issue with `onPreDraw` calls not getting called on multiple frames. The hope is that this PR fixes the issue in the 2nd PR as well as issue #1403 with text rendering.

*Update*
Confirmed with @gamepro65 that the 2 draw calls is expected when running the same `SampleView` on a device. First draw call is with the main `RecordedCanvas` and the second is a supplementary `RecordedCanvas` provided by [`RenderedThread`](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/view/ThreadedRenderer.java;l=740?q=ThreadedRenderer)
image

### CODE REVIEW

File: paparazzi-gradle-plugin/src/test/projects/compose/src/test/resources/compose_fonts.png
Comment (geoff-powell) [inline paparazzi-gradle-plugin/src/test/projects/compose/src/test/resources/compose_fonts.png:1]: it's interesting these files needed updating. I think the iguana layout lib caused a font rendering issue so we had updated these files and now we are undoing those updates.

File: paparazzi/src/test/java/app/cash/paparazzi/PaparazziTest.kt [hunk 1]
```diff
@@ -239,7 +241,7 @@ class PaparazziTest {
 
     paparazzi.gif(view, fps = 4)
 
-    assertThat(log).isEqualTo(listOf("draw", "predraw", "predraw", "predraw"))
+    assertThat(log).isEqualTo(listOf("predraw", "draw", "draw", "predraw", "predraw", "predraw"))
```
Comment (geoff-powell) [inline paparazzi/src/test/java/app/cash/paparazzi/PaparazziTest.kt:244]: unfortunately the fix here causes 2 draws to happen on the first frame.

**Comment by geoff-powell** [comment]
Unfortunately getting some issues on linux builds with my added test. What is strange is I pulled the generated image from the failed CI run and compared it to the one generated on my Mac ARM machine and the differ is less then 0.1 percent. But the test report says 3.3xxx%. 🤔 

Mac ARM generated image:
5e57e882d1350bc9d8be8b83fa9e84699f9affb9

Linux generated image:
5e57e882d1350bc9d8be8b83fa9e84699f9affb9

Delta image: (most likely due to platform drawing differences in `java.awt`
delta-5e57e882d1350bc9d8be8b83fa9e84699f9affb9

**Comment by geoff-powell** [comment]
Just noticed that other plugin tests sort files by `lastModified` so that seems to have fixed things. See in last commit.

**Comment by geoff-powell** [comment]
this change seems strange to have 2 `draw` calls occur but just confirmed in layout lib in jellyfish and iguana that the exact same thing happens there.

Screenshot 2024-05-09 at 3 58 08 PM

Using this sample view:
```
class SampleView(context: Context) : TextView(context) {

  init {
    text = "Hello, World!"

    viewTreeObserver.addOnPreDrawListener {
      text = "$text\n+OnPreDraw"
      true
    }
  }

  override fun onDraw(canvas: Canvas) {
    super.onDraw(canvas)
    text = "$text\n+OnDraw"
  }
}
```

File: paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt [hunk 2]
```diff
@@ -351,12 +351,6 @@ public class PaparazziSdk @JvmOverloads constructor(
   private fun createRenderSession(sessionParams: SessionParams): RenderSessionImpl {
     val renderSession = RenderSessionImpl(sessionParams)
     renderSession.setElapsedFrameTimeNanos(0L)
-    RenderSessionImpl::class.java
-      .getDeclaredField("mFirstFrameExecuted")
-      .apply {
-        isAccessible = true
-        set(renderSession, true)
-      }
```
[the remark this answers is not in the document]
Comment (gamepro65) [reply paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt:359]: Nice! Happy this is what it ended up being even with the extra onDraw side-effect. Mimicking what happens on device is best case scenario

**Review by gamepro65 — APPROVED** [review]
Thanks for diving into this!

**Comment by geoff-powell** [comment]
> Looks good! What ended up happening with the linux diff errors you were getting?

They issue was due to the order of snapshots generated. For some other plugin tests we are sorting by `lastModified` when asserting multiple snapshots. The rendering diff is platform specific with how layoutlib draws to the canvas. That is why we have the max percent difference of `0.1` as a default value

File: paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrodbx) [reply paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt:359]: Agreed!  Also, happy to have less hack code to work around this, so win-win!  One thing to watch out for over time is whether Layoutlib considers this extra side-effect and reverts.

Side note: will this extra draw call now "fix" the second frame issue we see on some Compose-based screenshots due to layout passes?  Or is that unrelated?

**Review by jrodbx — COMMENTED** [review]
makes sense.  nice investigation to ensure this is what we want!

File: paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (geoff-powell) [reply paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt:359]: nope, it does not fix that issue. I think that is unrelated and has to due with pending recompositions not executing in Paparazzi where they do run in layoutlib. Planning on taking a look into that this week actually

**Pull request merged**